src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -334,6 +323,16 @@ class PremisesController(
   ): ResponseEntity<Arrival> {
     val booking = getBookingForPremisesOrThrow(premisesId, bookingId)
 
+    if (booking.service == ServiceName.temporaryAccommodation.value) {
+      // TODO: Arrivals will likely need to check for overlaps once bed-level bookings are implemented for AP
+      val bedId = when (booking.bed) {
```

Comment:
Bikeshedding so feel free to ignore, but this might be a bit easier to read?

```
val bedId = booking.bed?.id 
      ?: throw InternalServerErrorProblem("No bed ID present on Temporary Accommodation booking: $bookingId")
```